dev/sg/release.go

```
@@ -0,0 +1,140 @@
+	}
+	hbPage := hbBuf.String()
+	if len(hbPage) == 0 {
+		return errors.New("provided reference uri does not have any contents")
+	}
+
+	var foundCVE []string
+	var unapprovedCVE []string
+
+	for _, artifact := range artifacts {
+		name := *artifact.Filename
+		url := *artifact.DownloadURL
+
+		pattern, err := regexp.Compile(`<\w+>(CVE.*)<\/\w+>`)
+		if err != nil {
+			return errors.Wrap(err, "failed to build regexp pattern")
+		}
+
+		if strings.HasSuffix(*artifact.Filename, "security-report.html") {
+			std.Out.WriteLine(output.Styledf(output.StylePending, "Checking security report: %s %s", name, url))
+
+			var buf bytes.Buffer
+			err = client.DownloadArtifact(artifact, &buf)
+			if err != nil {
+				return errors.Newf("failed to download security artifact %q at %s: %w", name, url, err)
+			}
+
+			matches := pattern.FindAllStringSubmatch(buf.String(), -1)
+			for _, match := range matches {
+				cve := strings.TrimSpace(match[1])
+				foundCVE = append(foundCVE, cve)
+				if !strings.Contains(hbPage, cve) {
+					unapprovedCVE = append(unapprovedCVE, cve)
+				}
+			}
+		}
+	}
+
+	// verbose := verboseFlag.Get(cmd)
+	std.Out.WriteLine(output.Styledf(output.StyleBold, "Found %d CVEs in the build", len(foundCVE)))
+	if verbose {
+		for _, s := range foundCVE {
+			std.Out.WriteLine(output.Styledf(output.StyleWarning, "%s", s))
+		}
+	}
+	if len(unapprovedCVE) > 0 {
+		std.Out.WriteLine(output.Styledf(output.StyleWarning, "Unable to match CVEs"))
+		for _, s := range unapprovedCVE {
+			std.Out.WriteLine(output.Styledf(output.StyleWarning, "%s", s))
+		}
+	} else {
+		std.Out.WriteLine(output.Styledf(output.StyleSuccess, "All CVEs approved!"))
+	}
+
+	return nil
+}
+
+func downloadUrl(uri string, w io.Writer) (err error) {
+	std.Out.WriteLine(output.Styledf(output.StylePending, "Downloading url: %s", uri))
+	client := http.Client{}
+	resp, err := client.Get(uri)
+	if err != nil {
+		return err
+	}
+	defer func() { err = resp.Body.Close() }()
```

this will always clobber an error from `io.Copy` below, right?

---

oops - you're right, I'll fix that

---

Seems like it's a standard pattern to ignore the error in Close, so I've just done that https://github.com/sourcegraph/sourcegraph/pull/44585/commits/225b814d3c117c54b9e75a79bc55c0173380e472